pycaret/containers/models/classification.py

```diff
@@ -1147,8 +1147,15 @@ def __init__(self, experiment):
             "n_jobs": experiment.n_jobs_param,
             "verbosity": 0,
             "booster": "gbtree",
-            "tree_method": "gpu_hist" if experiment.gpu_param else "auto",
         }
+
+        # If XGBoost >= 2 change and add new parameters
+        if version.parse(xgboost.__version__) >= version.parse("2.0.0"):
+            args["tree_method"] = "hist"
+            args["device"] = "cuda"
```

Comment:
Following the [docs](https://xgboost.readthedocs.io/en/stable/parameter.html#general-parameters), `device` should not be `cuda`, but this instead `"gpu" if experiment.gpu_param else "cpu"`.

Reply:
Here, as I have a laptop with an nvida graphics card, I had no problems, but I'm not sure what happens for different systems, for example amd or intel. I think it looks for compatibility or switches to cpu automatically.
@trivialfis can provide extra help...

Reply:
At the moment, CUDA is the only supported platform for GPU.

I worked for NVIDIA, but XGBoost accepts PRs for other platforms if there are effort to work on it along with CI infrastructure.

Reply:
Thanks. In that case I think it's still better to use `"gpu" if experiment.gpu_param else "cpu"`, since it's robust to future GPU implemnetations, and will still default to `cuda` at the moment. Using `cuda` will force cuda in the future and fail if only OneAPI is available. Also better to specify `cpu` if `experiment.gpu_param=False` since explicit is better than implicit.